src/spyglass/common/common_subject.py

```diff
@@ -18,23 +18,70 @@ class Subject(SpyglassMixin, dj.Manual):
     """
 
     @classmethod
-    def insert_from_nwbfile(cls, nwbf):
-        """Get the subject info from the NWBFile, insert into the Subject."""
-        sub = nwbf.subject
-        if sub is None:
+    def insert_from_nwbfile(cls, nwbf, config):
+        """Get the subject info from the NWBFile, insert into the Subject.
+
+        Parameters
+        ----------
+        nwbf: pynwb.NWBFile
+            The NWB file with subject information.
+        config : dict
+            Dictionary read from a user-defined YAML file containing values to
+            replace in the NWB file.
+
+        Returns
+        -------
+        subject_id : string
+            The id of the subject found in the NWB or config file, or None.
+        """
+        if "Subject" not in config and nwbf.subject is None:
             logger.warn("No subject metadata found.\n")
-            return
+            return None
+
+        conf = config["Subject"][0] if "Subject" in config else dict()
+        sub = (
+            nwbf.subject
+            if nwbf.subject is not None
+            else type("DefaultObject", (), {})()
+        )
+
```

Comment:
I think you could simplify here with some of python's `get` methods that return defaults when the item isn't found
- `my_dict.get('key', 'default_val')` accepts a default, but will pass `None` if not provided
- `getattr(my_class, 'my_attr', 'default_val')` requires you to specify a default

here's what I came to after running it all through a dict comprehension...
```python
subject_dict = {
    field: conf.get(field, getattr(sub, field, None))
    for field in [
        "subject_id",
        "age",
        "description",
        "genotype",
        "species",
        "sex",
    ]
}
if (sex := subject_dict["sex"][0].upper()) in ("M", "F"):
    subject_dict["sex"] = sex
else:
    subject_dict["sex"] = "U"
```

Reply:
Updated! Thanks